clang/test/Driver/linker-wrapper-sycl.cpp

```diff
@@ -47,11 +47,6 @@
 // RUN: clang-linker-wrapper -sycl-device-libraries=%t.devicelib.o -sycl-post-link-options="SYCL_POST_LINK_OPTIONS" -llvm-spirv-options="LLVM_SPIRV_OPTIONS" "--host-triple=x86_64-unknown-linux-gnu" "--triple=spir64" "--linker-path=/usr/bin/ld" -shared "--" HOST_LINKER_FLAGS "-dynamic-linker" HOST_DYN_LIB "-o" "a.out" HOST_LIB_PATH HOST_STAT_LIB %t.o --dry-run 2>&1 | FileCheck -check-prefix=CHK-SHARED %s
 // CHK-SHARED: "{{.*}}clang"{{.*}} -fPIC
 
-// RUN: rm %T/linker_wrapper_dump || true
-// RUN: clang-linker-wrapper -sycl-dump-device-code=%T/linker_wrapper_dump -sycl-device-libraries=%t.devicelib.o "--host-triple=x86_64-unknown-linux-gnu" "--triple=spir64" "--linker-path=/usr/bin/ld" -shared "--" HOST_LINKER_FLAGS "-dynamic-linker" HOST_DYN_LIB "-o" "a.out" HOST_LIB_PATH HOST_STAT_LIB %t.o --dry-run
```

Comment:
Shouldn't we leave that?

Reply:
No. This test is utterly broken. Let me know if you have suggestions on writing a working test.

Reply:
I see. The following piece of code should be reworked so that it uses an empty `MemoryBuffer` as you suggested in an internal tracker.
https://github.com/intel/llvm/blob/5c5954eb81ce04a84a5d3757632630758ef9578d/clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp#L1074-L1086

Then this test could be returned.

Reply:
It will also require significant test rework as it checks that files are created in the file system in dry-run mode.
I suggest you rework the dry-run mode for all SYCL methods and all corresponding tests in separate PR.
Does it make sense to you?